compat/compat.php

```diff
@@ -15,19 +16,32 @@ public static function single() {
 	}
 
 	function __construct() {
-		add_action('wp', array( $this, 'init' ), 9 );
+		$builder = $this->get_active_builder();
+		if ( ! empty( $builder ) ) {
+			require_once $builder['file_path'];
+		}
 	}
 
-	function init() {
-		if ( $this->is_active( self::BEAVER_BUILDER ) ) {
-			require_once plugin_dir_path( __FILE__ ) . 'beaver-builder/beaver-builder.php';
+	function get_active_builder() {
+
+		$builders = include_once 'builders.php';
+
+		foreach ( $builders as $builder ) {
+			if ( $this->is_active( $builder ) ) {
+				return $builder;
+			}
 		}
+
+		return null;
 	}
 
 	function is_active( $builder ) {
-		switch ( $builder ) {
+		switch ( $builder[ 'name' ] ) {
 			case self::BEAVER_BUILDER:
-				return class_exists( 'FLBuilderModel' ) && FLBuilderModel::is_builder_active();
+				return class_exists( 'FLBuilderModel' );
+			break;
+			case self::ELEMENTOR:
+				return version_compare( PHP_VERSION, '5.4', '>=' ) && class_exists( 'Elementor\\Plugin' );
```

Comment:
You don't need to compare PHP version. Just check if `\Elementor\Plugin` class exists. We don't include any files if the website is not in our [requirements](https://docs.elementor.com/article/38-requirements).